src/client/app/components/BarControlsComponent.tsx

```diff
@@ -17,26 +16,81 @@ import TooltipMarkerComponent from './TooltipMarkerComponent';
  */
 export default function BarControlsComponent() {
 	const dispatch = useAppDispatch();
+
+	// The min/max days allowed for user selection
+	const MIN_BAR_DAYS = 1;
+	const MAX_BAR_DAYS = 366;
+	// Special value if custom input for standard menu.
+	const CUSTOM_INPUT = '-99';
+
+	// This is the current bar interval for graphic.
 	const barDuration = useAppSelector(selectBarWidthDays);
 	const barStacking = useAppSelector(selectBarStacking);
-	const [showSlider, setShowSlider] = React.useState<boolean>(false);
-	const [sliderVal, setSliderVal] = React.useState<number>(barDuration.asDays());
-
-	const toggleSlider = () => {
-		setShowSlider(showSlider => !showSlider);
-	};
+	// Holds the value of standard bar duration choices used so decoupled from custom and
+	// also to allow special value for custom choice.
+	const [barDays, setBarDays] = React.useState<string>(barDuration.asDays().toString());
+	// Holds the value during custom bar duration input so only update graphic when done entering and
+	// separate from standard choices.
+	const [barDaysCustom, setBarDaysCustom] = React.useState<number>(barDuration.asDays());
+	// True if custom bar duration input is active.
+	const [showCustomBarDuration, setShowCustomBarDuration] = React.useState<boolean>(false);
 
 	const handleChangeBarStacking = () => {
 		dispatch(graphSlice.actions.changeBarStacking());
 	};
 
-	const handleSliderChange = (value: number) => {
-		setSliderVal(value);
+	// Update when the custom input bar duration is shown/hidden.
+	React.useEffect(() => {
+		if (showCustomBarDuration) {
+			// It changed to true so set the input value as the current barDuration
+			// so shows in custom field at start of input.
+			setBarDaysCustom(barDuration.asDays());
+		}
+		// If false then hiding custom so value does not matter.
+	}, [showCustomBarDuration]);
+
+	// Returns true if this is a valid bar duration.
+	const barDaysValid = (barDays: number) => {
+		return Number.isInteger(barDays) && barDays >= MIN_BAR_DAYS && barDays <= MAX_BAR_DAYS;
 	};
+
+	// Updates values when the standard bar duration menu is used.
+	const handleBarDaysChange = (value: string) => {
+		if (value === CUSTOM_INPUT) {
+			// Set menu value for standard bar to special value to show custom
+			// and show the custom input area.
+			setBarDays(CUSTOM_INPUT);
+			setShowCustomBarDuration(true);
+		} else {
+			// Set bar duration for graphing, the standard menu value
+			// and hide the custom bar duration input.
+			// Since controlled values know it is a valid integer.
+			console.log('setting barDuration to ', Number(value));
+			updateBarDurationChange(Number(value));
+			setBarDays(value);
+			setShowCustomBarDuration(false);
```

Comment:
Minor issue when reverting from large bar width days to smaller there can be a bit of lag when plotly renders (from 4 weeks to 1 day for example causes a few seconds of lag on my machine). During this time it is still redering custom input and associated input box. This can be remedied calling updateBardurationChange last, so
 setBarDays(...); 
 setShowCustom(...); 
 updateBarDurationChange(...);